source/read-write-concern/tests/operation/default-write-concern-3.2.yml

```diff
@@ -0,0 +1,125 @@
+# Test that setting a default write concern does not add a write concern
+# to the command sent over the wire.
+# Test operations that require 3.2+ server, where findAndModify started
+# to accept a write concern.
+
+data:
+  - {_id: 1, x: 11}
+  - {_id: 2, x: 22}
+collection_name: &collection_name default_write_concern_coll
+database_name: &database_name default_write_concern_db
+other_collection_name: &other_collection_name other_collection_name
+
+runOn:
+    - minServerVersion: "3.2"
+
+tests:
+  - description: 'findAndModify operations omit default write concern'
+    clientOptions: {writeConcern: {}}
+    operations:
+      - name: findOneAndUpdate
+        object: collection
+        databaseOptions: {writeConcern: {}}
+        collectionOptions: {writeConcern: {}}
```

Comment:
What is the logic in applying a WC at all three levels? Presumably the collection option would overwrite any others, so isn't this redundant? I think this is definitely the case with database and collection options, although I imagine libmongoc might handle client-level WC separately.

Reply:
Yeah it is redundant. The collection write concern should overwrite. I suppose I was thinking a bug in the inheritance logic could result in differing behavior, so that it'd be safer to specify it everywhere. But I don't think it's a strong case. I've changed it to only apply at the collection or database level depending on the operation.